### PR TITLE
Task#197609: Show user list: waiting, reading, returning, returned

### PR DESCRIPTION
**Pull request by HaiHaChan**
https://dev.framgia.com/issues/197609

### CODE REVIEW

File: app/Repositories/Eloquents/BookUserEloquentRepository.php [hunk 1]
```diff
@@ -69,4 +69,20 @@ public function getDataRequest($data = [], $with = [], $dataSelect = ['*'], $att
...
+            $with = [
+                'user',
+            ];
+
+            return $this->getData($request, $with)
+                ->pluck('user')
+                ->unique()
+                ->chunk(config('view.paginate.follow_user'));
+        } catch (Exception $e) {
+            return NULL;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookUserEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.PHP.LowerCaseConstant.Found</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BookUserEloquentRepository.php</br>TRUE, FALSE and NULL must be lowercase; expected "null" but found "NULL"

File: app/Http/Controllers/User/BookController.php [hunk 2]
```diff
@@ -267,4 +271,21 @@ public function destroy($id)
...
+        ];
+        $data = $this->bookUser->getDetailData($req);
+        if (Auth::check()) {
+            $followingIds = Auth::user()->followings->pluck('id')->toArray();
+        } else {
+            $followingIds = NULL;
+        }
+        $type = $request->type;
+
+        return view('layout.section.user_list', compact('data', 'type', 'followingIds'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/BookController.php:288]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.PHP.LowerCaseConstant.Found</br>**Severity**: error</br>**File**: app/Http/Controllers/User/BookController.php</br>TRUE, FALSE and NULL must be lowercase; expected "null" but found "NULL"

Comment (dangminhtruong) [reply app/Http/Controllers/User/BookController.php:288]: @framgiaciv3 (clap)

Comment (BunnyPi04) [reply app/Http/Controllers/User/BookController.php:288]: @dangminhtruong anh check sao nó báo lỗi chỗ này đi anh

Comment (dangminhtruong) [reply app/Http/Controllers/User/BookController.php:288]: @BunnyPi04  Xem trên trang CIV3 cho chính xác em ạ. Con này chưa ổn định, đợi có thời gian bọn anh sửa lại.... 
File này bị lỗi ở đây thì phải này https://civ3-dev.framgia.vn/repositories/PHPEducation/Fbook/builds/11/violation -> phpcs -> app/Http/Controllers/User/BookController.php

Comment (HaiHaChan) [reply app/Http/Controllers/User/BookController.php:288]: dạ em vừa fix lại rồi ạ. mà k biết có được không :(

File: app/Repositories/Eloquents/BookUserEloquentRepository.php [hunk 3]
```diff
@@ -51,7 +51,7 @@ public function UpdateBookRequest($data)
         $bookRequest = $this->model()->findOrFail($data['id']);
         if (isset($data['status']) && $data['status'] == config('view.request.waiting')) {
             $type['type'] = 'reading';
-        } elseif (isset($data['status']) && $data['status'] == config('view.request.reading') ||$data['status'] == config('view.request.returning')) {
+        } elseif (isset($data['status']) && $data['status'] == config('view.request.reading') || $data['status'] == config('view.request.returning')) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BookUserEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Repositories/Eloquents/BookUserEloquentRepository.php</br>Line exceeds 120 characters; contains 151 characters

File: app/Repositories/Eloquents/FollowEloquentRepository.php [hunk 4]
```diff
@@ -31,7 +31,7 @@ public function store($data = [])
         }
     }
 
-     public function destroy($data){
+    public function destroy($data){
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/FollowEloquentRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/FollowEloquentRepository.php</br>Opening brace should be on a new line

File: app/Http/Middleware/AdminMiddleware.php [hunk 5]
```diff
@@ -26,7 +26,8 @@ public function handle($request, Closure $next)
                             return $next($request);
                         }
                     }
-                } elseif ($roles[0]->name == config('view.role.admin') || $roles[0]->name == config('view.role.editor')) {
+                } elseif ($roles[0]->name == config('view.role.admin')
```
Comment (framgiaciv3) [inline app/Http/Middleware/AdminMiddleware.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Middleware/AdminMiddleware.php</br>Whitespace found at end of line

File: app/Http/Controllers/User/BookController.php (hunk 2 above)
Comment (dangminhtruong) [reply app/Http/Controllers/User/BookController.php:288]: @HaiHaChan Bé fix hết chỗ lỗi phpcs trong này nè em https://civ3-dev.framgia.vn/repositories/PHPEducation/Fbook/builds/13/violation

Comment (dangminhtruong) [reply app/Http/Controllers/User/BookController.php:288]: Add chatwork anh chụp hình gửi cho nhé...

**Pull request merged**